src/manipulation.js

```diff
@@ -460,7 +460,11 @@ jQuery.fn.extend({
 		});
 
 		// Force removal if there was no new content (e.g., from empty arguments)
-		return arg && (arg.length || arg.nodeType) ? this : this.remove();
+		if ( !arg || !arg[ 0 ] || typeof arg === "string" ) {
```

Comment:
Or

```js
if ( arg && (arg[ 0 ] || arg.nodeType) && typeof arg !== "string" ) {
	return this;
}

// Force removal if there was no new content (e.g., from empty arguments)
return this.remove();
```

Which i actually like better, since `remove` would not be called as much